src/test/java/org/opentripplanner/middleware/persistence/TripHistoryPersistenceTest.java

```diff
@@ -0,0 +1,187 @@
+package org.opentripplanner.middleware.persistence;
+
+import com.mongodb.client.model.Filters;
+import org.bson.conversions.Bson;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.middleware.OtpMiddlewareTest;
+import org.opentripplanner.middleware.models.TripRequest;
+import org.opentripplanner.middleware.models.TripSummary;
+import org.opentripplanner.middleware.models.OtpUser;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.List;
+
+import static com.mongodb.client.model.Filters.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.opentripplanner.middleware.persistence.PersistenceUtil.*;
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsText;
+
+/**
+ * Tests to verify that trip request and trip summary persistence in MongoDB collections are functioning properly. A
+ * number of {@link TypedPersistence} methods are tested here, but the HTTP endpoints defined in
+ * {@link org.opentripplanner.middleware.controllers.api.ApiController} are not themselves tested here.
+ */
+public class TripHistoryPersistenceTest extends OtpMiddlewareTest {
+    private static final String TEST_EMAIL = "john.doe@example.com";
+    private static final String OTP_SERVER = getConfigPropertyAsText("OTP_SERVER");
+    private static final String OTP_SERVER_PLAN_END_POINT = getConfigPropertyAsText("OTP_SERVER_PLAN_END_POINT");
+
+    TripRequest tripRequest = null;
+    TripSummary tripSummary = null;
+    List<TripRequest> tripRequests = null;
+
+    @Test
+    public void canCreateTripRequest() {
+        String userId = "123456";
+        tripRequest = createTripRequest(userId);
+        String id = tripRequest.id;
+        TripRequest retrieved = Persistence.tripRequests.getById(id);
+        assertEquals(id, retrieved.id, "Found Trip request ID should equal inserted ID.");
+    }
+
+    @Test
+    public void canDeleteTripRequest() {
+        String userId = "123456";
+        TripRequest tripRequestToDelete = createTripRequest(userId);
+        Persistence.tripRequests.removeById(tripRequestToDelete.id);
+        TripRequest tripRequest = Persistence.tripRequests.getById(tripRequestToDelete.id);
+        assertNull(tripRequest, "Deleted TripRequest should no longer exist in database (should return as null).");
+    }
+
+    @Test
+    public void canCreateTripSummaryWithError() {
+        tripSummary = createTripSummaryWithError(OTP_SERVER, OTP_SERVER_PLAN_END_POINT);
```

Comment:
Oh yikes, sorry for not seeing this earlier, but we should be very careful about making HTTP requests during tests. Since this is merely a persistence test, we don't need to make an HTTP request. Instead, we should create a mock trip summary and save that. This has the added benefit of making the test faster and having it not fail if for some reason the HTTP request is unsuccessful.

Ideally, we'd also test the OtpDispatcher in a separate test file and use wiremock (see [this example of using wiremock](https://github.com/ibi-group/datatools-server/blob/c691cf6f22159b2cbeee45569c73b1777e1db1dc/src/test/java/com/conveyal/datatools/manager/controllers/api/UserControllerTest.java#L89) in datatools-server) to mock HTTP requests/repsonses.

Reply:
@evansiroky - The call to the OTP plan endpoint was just to obtain a plan response. I have now created static files containing these instead. This has rendered OtpDispatcherHelper.java redundant so have removed it.